src/main/java/com/mykaarma/urlshortener/util/UrlServiceUtil.java

```diff
@@ -219,5 +221,35 @@ public String generateValidHash(int hashLength){
 		log.info("Time taken to generate a hash {} is {}ms",shortUrlHash, System.currentTimeMillis()-t1);
 		return shortUrlHash;
 	}
+
+	/**
+	 * This function generates key which will be used by redis for obtain lock.
+	 * Encoding of URL is done to ensure that there are no characters that could potentially interfere with key naming conventions in Redis
+	 * @param businessUuid
+	 * @param longUrl
+	 * @param domainName
+	 * @return lockKey
+	 * @throws Exception
+	 */
+
+	public String getRedisKeyForCreateShortUrl(String businessUuid, String longUrl, String domainName) {
```

Comment:
can be a static function

Reply:
good suggestion. made it a static function.